trapi_throttle/server.py

```diff
@@ -119,26 +122,69 @@ async def process_batch(kp_id):
         # Process batch
         batch_request_ids = [key.split(':')[-1] for key in batch_keys]
 
-        request_values_db = [
-            RedisValue(APP.state.redis, f"{kp_id}:buffer:{request_id}")
+        request_values_db_get = {
+            request_id : RedisValue(APP.state.redis, f"{kp_id}:buffer:{request_id}").get()
             for request_id in batch_request_ids
-        ]
-        request_values = await asyncio.gather(*[
-            request_value_db.get() for request_value_db in request_values_db
-        ])
+        }
+        request_value_mapping = await gather_dict(request_values_db_get)
+
+        # Extract a curie mapping from each request
+        request_curie_mapping = {
+            request_id: extract_curies(request_value["message"]["query_graph"])
+            for request_id, request_value in request_value_mapping.items()
+        }
+
+        # Find requests that are the same (those that we can merge)
+        # This disregards non-matching IDs because the IDs have been
+        # removed with the extract_curie method
+        request_value_mapping = get_equal_dict_values(request_value_mapping)
+
+        # Filter curie mapping to only include matching requests
+        request_curie_mapping = {
+            k:v for k,v in request_curie_mapping.items()
```

Comment:
Please run pycodestyle for PEP8 compliance.
```suggestion
            k: v for k, v in request_curie_mapping.items()
```